### PR TITLE
ci: Double tsan CPU and Memory to avoid global timeout

### LINKED ISSUE #19864: Spurious CI failure: cirrus CI timeout
Not sure if this is the right place to report it, but I saw this job failure today:
https://cirrus-ci.com/task/6078479919742976

I'll try re-running but opening this issue in case others see this kind of thing too.

### PR DESCRIPTION
**Pull request by maflcko**
Fix #19864

### CODE REVIEW

**Comment by MarcoFalke** [comment]
https://github.com/bitcoin/bitcoin/blob/416efcb7ab4545d65c675338065aaaa46f800748/.cirrus.yml#L3

File: .cirrus.yml [hunk 1]
```diff
@@ -56,7 +56,10 @@ task:
   << : *GLOBAL_TASK_TEMPLATE
   container:
     image: ubuntu:focal
+    cpu: 4  # Double CPU and Memory to avoid timeout
+    memory: 16G
   env:
+    MAKEJOBS: "-j8"
```
Comment (hebasto) [inline .cirrus.yml:62]: `cpu: 4`, `MAKEJOBS: "-j8"` and `memory: 16G` are interconnected. Is there any justification that that combination is optimal?

**Review by hebasto — COMMENTED** [review]
Concept ACK.

File: .cirrus.yml (hunk 1 above)
Comment (MarcoFalke) [reply .cirrus.yml:62]: MAKEJOBS is also used to run the functional tests in ci in parallel. To use maximum CPU during functional tests, at least num_cpu * 2 needs to be passed.

```
$ ./test/functional/test_runner.py --help|grep parallel
  --jobs JOBS, -j JOBS  how many test scripts to run in parallel. Default=4.

Comment (hebasto) [reply .cirrus.yml:62]: > To use maximum CPU during functional tests, at least num_cpu * 2 needs to be passed.

TIL.

Comment (MarcoFalke) [reply .cirrus.yml:62]: At least this is what I found out a while back. Might have changed in the meantime

**Review by hebasto — APPROVED** [review]
ACK fa8e1487144eab237ffd291397355ef4801f46f8, according to https://cirrus-ci.org/guide/linux/ the limits are:
- 8 CPUs
- 24 GB of memory

**Comment by practicalswift** [comment]
ACK fa8e1487144eab237ffd291397355ef4801f46f8 -- patch looks correct

**Pull request merged**